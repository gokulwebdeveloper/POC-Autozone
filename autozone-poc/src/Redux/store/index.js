import {createStore, applyMiddleware, compose} from 'redux';
import rootReducer from '../reducers';
import rootSaga from '../middlewares/saga';
import createSagaMiddleware from 'redux-saga';

// create the saga middleware
export const sagaMiddleware = createSagaMiddleware();

//list of middlewares
const middlewares = [sagaMiddleware];

//apply middleware on store
const store = createStore(rootReducer, applyMiddleware(...middlewares));

//run saga generator function
sagaMiddleware.run(rootSaga);

export default store;