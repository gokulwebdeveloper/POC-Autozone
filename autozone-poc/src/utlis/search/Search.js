import React, {Fragment} from "react";


const Search = () => {
    return (
        <Fragment>
            <div className="search-area">
                <form action="#" method="post">
                    <input type="search" name="search" id="headerSearch" placeholder="Type for search" />
                    <button type="submit"><i className="fa fa-search" aria-hidden="true"></i></button>
                </form>
            </div>
        </Fragment>
    )
}

export default Search;  