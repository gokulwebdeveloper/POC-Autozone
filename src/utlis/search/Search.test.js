import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from './Search';
import { Provider } from 'react-redux';
import store from '../../Redux/store';

configure({ adapter: new Adapter() });

describe('Search testing', () => {
    let wrapper = shallow(<Provider store={store}><Search /></Provider>);

    it("SnapShot Match for Search component", () => {
        const page = renderer.create(<Provider store={store}><Search /></Provider>).toJSON();
        expect(page).toMatchSnapshot();
    })

    it("displays search bar", () => {
        const inputType = wrapper.find("input");
        expect(inputType).toBeTruthy();
    });

});