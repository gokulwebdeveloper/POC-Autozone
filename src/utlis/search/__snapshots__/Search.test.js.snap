// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search testing SnapShot Match for Search component 1`] = `
<div
  className="search-area"
>
  <input
    id="headerSearch"
    name="search"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type for search"
    type="search"
  />
  <button>
    <i
      aria-hidden="true"
      className="fa fa-search"
    />
  </button>
</div>
`;
